### PR TITLE
bug 1271509: Add scrape_user command and framework

### PR DESCRIPTION
**Pull request by jwhitlock**
Add a content scraping framework, for adding or updating data to a local Kuma instance. The first user of the framework is the management command ``scrape_user``, which can mirror a production user locally, using only public data on the user's profile.

Some things to try:

    ./manage.py scrape_user username
    ./manage.py scrape_user username --force
    ./manage.py scrape_user username --force --social
    ./manage.py scrape_user username --force --social --email name@example.com

Some of the framework may seem like overkill, but it's needed for the much more complex task of scraping a document, coming in the next PR.

### CODE REVIEW

**Comment by jwhitlock** [comment]
The scraper doesn't handle banned users after we changed from a 403 to a 404. New commit incoming...

File: docs/data.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- Search filters and tags (but not a search index, which must be created
+  locally - see :ref:`indexing-documents`)
+- The Mozilla Hacks feed
+- Test users, with appropriate groups and permissions:
+
+  - ``test-super`` - A user with full permissions
+  - ``test-moderator`` - A staff content moderator
+  - ``test-new`` - A regular user account
+  - ``test-banned`` - A banned user
+  - ``viagra-test-123`` - An unbanned spammer
```
Comment (escattone) [inline docs/data.rst]: 😆

Comment (jwhitlock) [inline docs/data.rst]: I can't take full credit for this one - it's from the [Akismet docs](https://akismet.com/development/api/#detailed-docs), and it used to always mark a comment as spam

File: kuma/scrape/scraper.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+                # Gather dependent sources
+                source_num += 1
+                old_state = source.state
+                dependencies = source.gather(self.requester, self.storage)
+                new_sources.extend(dependencies)
+                dep_count = len(dependencies)
+                state_counts[source.state] += 1
+                for dep in dependencies:
+                    if "%" in dep[1]:
```
Comment (escattone) [inline kuma/scrape/scraper.py]: It's not clear to me why this is needed, but perhaps it'll be more clear in the follow-on PR's?

File: kuma/scrape/tests/test_scraper.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+#
+
+
+def test_session():
+    """A session is created once on request."""
+    requester = Requester('example.com', True)
+    session = requester.session
+    assert isinstance(session, requests.Session)
+    session2 = requester.session
+    assert session == session2
```
Comment (escattone) [inline kuma/scrape/tests/test_scraper.py]: It seems this could be even stronger, ``assert session is session2``?

Comment (jwhitlock) [inline kuma/scrape/tests/test_scraper.py]: 👍 that works, will update

**Comment by jwhitlock** [comment]
Thanks for the quick review!  The next one will be ``scrape_document``, and will be n-a-s-t-y.  Aiming for Friday.

**Pull request merged**